### PR TITLE
[opencv] Add opencv@3

### PR DESCRIPTION
**Pull request by purcell**
OpenCV was recently bumped to 4.x in #35521, but a lot of non-Homebrew-packaged code does not yet compile against 4.x, which was only released just over a month ago.

Therefore, since there is an `opencv@2` formula, it also seems prudent to provide `opencv@3`. This PR does that, based on the most recent version of `opencv.rb` for OpenCV 3.x.

-----

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by SMillerDev** [comment]
The tests fail on Jenkins, coup you check those?

**Comment by purcell** [comment]
Thanks. I managed to reproduce this locally, and have pushed a fix: 79c7aefa536eebc9bb362c345276270dc5e5e82a

I determined this by noting that the formula installed its Python modules into directories with the following names:
```
/usr/local/Cellar/opencv@3/3.4.5/lib/python2.7/ (6 files)
/usr/local/Cellar/opencv@3/3.4.5/lib/python3.7/ (6 files)
```

**Comment by fxcoudert** [comment]
For what it's worth: the 3 formulas in Homebrew core that depend on OpenCV could be adapted to OpenCV 4 by a simple patch, which was already available upstream in most cases. The breaking changes appear limited.

The criteria for new versions formulas in Homebrew core are here https://docs.brew.sh/Versions. They include:

> Upstream should have a release branch for the versioned formulae version and still make security updates for that version

So: does the OpenCV project maintain the 3.x branch, and plans to make future releases?

File: Formula/opencv@3.rb [hunk 1]
```diff
@@ -0,0 +1,120 @@
+class OpencvAT3 < Formula
+  desc "Open source computer vision library"
+  homepage "https://opencv.org/"
+  url "https://github.com/opencv/opencv/archive/3.4.5.tar.gz"
+  sha256 "0c57d9dd6d30cbffe68a09b03f4bebe773ee44dc8ff5cd6eaeb7f4d5ef3b428e"
+
+  bottle do
```
Comment (fxcoudert) [inline Formula/opencv@3.rb]: No bottle block, please.

Comment (purcell) [reply Formula/opencv@3.rb]: > No bottle block, please.

Removed in e1f9843e7

**Comment by purcell** [comment]
> So: does the OpenCV project maintain the 3.x branch, and plans to make future releases?

Yes:

Branch 3.4 will be switched to maintanence mode: only bugfixes and light features will be accepted. BTW, release 3.4.4 is ready too!

**Comment by purcell** [comment]
> For what it's worth: the 3 formulas in Homebrew core that depend on OpenCV could be adapted to OpenCV 4 by a simple patch, which was already available upstream in most cases. The breaking changes appear limited.

Agreed, but people also use Homebrew as a source of native libraries for non-Homebrew work, so bridging changes like this can be helpful. In my case, I'm using language bindings for OpenCV that are not yet compatible with OpenCV 4. With any luck, I'll get them updated, but it's still only a matter of weeks since the OpenCV release. Obviously after a few months it would be best to drop these versioned formulae where possible: there may well be a case for dropping `opencv@2` immediately.

**Comment by purcell** [comment]
Yep, understandably difficult to strike the right balance. You folks seem to do a wonderful job, though, so thanks!

**Comment by neggert** [comment]
FWIW, there's at least one place where the 4.x release does not maintain feature parity with 3.x. Quoting from the [changelog](https://github.com/opencv/opencv/wiki/ChangeLog#version400):

> For now base64 support is not complete (only loading base64-encoded XML and YAML is supported, encoding is not supported at all)

I'm not going to be able to migrate my code until they fix this, so I'd really appreciate having `opencv@3` available.

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
That all sounds reasonable, let's go for it!

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @purcell for your contribution to Homebrew!

**Comment by purcell** [comment]
> Thanks @purcell for your contribution to Homebrew!

Cheers @fxcoudert, your help and work on homebrew is appreciated.

**Pull request closed without merging**